src/Image/Image.php

```diff
@@ -337,9 +336,9 @@ public function output(string $type, int $quality = 75)
 
     /**
      * @param  string  $path
-     * @param $type
+     * @param  string  $type
      * @param  int  $quality
-     * @return false|null|string
+     * @return ($path is null ? string|false : void)
```

Comment:
It is not a valid [Type](https://docs.phpdoc.org/3.0/guide/references/phpdoc/types.html) no.

Reply:
If path is null it returns the image blob and if path is provided it tries to save file to the path and returns void.

Reply:
> If path is null it returns the image blob

So in the current implementation `save` only saves if the `$path` is given and the `$type` is valid? Otherwise, it returns the blob for the user to save it themselves. So it doubles as a `getImage` method as well? Only it's not documented it has this behavior. Should `save` be responsible for this behavior though? Seems `save` is used as a catch-all to manage multiple different responsibilities?

Reply:
@stnguyen90 it is a conditional return type https://phpstan.org/writing-php-code/phpdoc-types#conditional-return-types